### ISSUE TITLE
Refactor(Action Bar): match Figma and utilize Tailwind

### ISSUE THREAD

**Issue by julio8a** [Esri/calcite-components#2452]
Related prs:
- pr #1805
- pr #2429

5/19 QA: the only additional changes needed:
- upping margin-right for .button--text-visible .icon-container from mr-2 to mr-3:
https://github.com/Esri/calcite-components/blob/master/src/components/calcite-action/calcite-action.scss#L110
- ensure font-size matches Figma (may need `h` suffix)

**Comment by asangma** [comment]
ActionBar doesn't support scale.

I believe the extra space around the trigger was addressed [here](https://github.com/Esri/calcite-components/pull/2557).

> There is white space around the action-menu items on hover. This was due to accessibility hover/focus states.

Is this a suggestion for a code change or a Figma update?

**Comment by bstifle** [comment]
Why would we need to do that with the hover state, to account for the focus ring...?

The hover/press states should go edge to edge

**Comment by julio8a** [comment]
Same comment I posted in the action-pad, currently, action-bar doesn't support scale, which we need to add (S and L). It's really odd how it works now where you can change the scale of `action` inside action bar and it breaks.

The goal would be to set it at the parent level (action-bar) and action would follow, in this instance.

**Comment by macandcheese** [comment]
Is there a way to address this at the action level? Why does the focus ring rely on space around the element?

This 1px space occurs when using calcite-action elements that are slotted (in this case, in a panel). Focus, hover, placement all seem to have slightly different visuals, and looks sloppy. Can we eliminate these small px differences with action?

<img width="800" alt="Screen Shot 2021-08-11 at 9 46 37 AM" src="https://user-images.githubusercontent.com/4733155/129070316-1fc9ff35-5f24-4bd7-ae1c-acf1f841b058.png">
<img width="800" alt="Screen Shot 2021-08-11 at 9 49 26 AM" src="https://user-images.githubusercontent.com/4733155/129070310-fdd7e410-524b-4632-a454-6df81fe1c387.png">
<img width="800" alt="Screen Shot 2021-08-11 at 9 46 21 AM" src="https://user-images.githubusercontent.com/4733155/129070313-26d169f8-3063-45b4-b636-8c04abf59664.png">
<img width="800" alt="Screen Shot 2021-08-11 at 9 49 15 AM" src="https://user-images.githubusercontent.com/4733155/129070298-9c237c59-6164-4697-811f-843306421d50.png">

**Comment by bstifle** [comment]
agreed the margin is strange. things should be edge to edge. focus should just be an internal 2px stroke

**Comment by asangma** [comment]
In ActionMenu, the button inside Action is not actually getting focused. That's why the additional fun tims. The Action is getting `active` set on it, and the focus styles are targeted to slotted Actions with that prop.

@driskull Is there a straight-forward way for ActionMenu to tell its slotted Actions to handle focus (i.e. pass focus on to the button in Action's shadow DOM)?

**Comment by driskull** [comment]
@asangma unfortunately, no. The actions should not actually be focused so we need to have active styles that mimmic it better. 

We're going to run into this more and more as we only allow focus on one element at a time for a component. Especially if the component opens a menu or something like that.

https://www.digitala11y.com/aria-activedescendant-properties/
https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-actions-active-descendant.html

**Comment by driskull** [comment]
Here's a good summary: https://developers.google.com/web/fundamentals/accessibility/focus/using-tabindex#managing_focus_in_components

**Comment by asangma** [comment]
Thanks @driskull.
@bstifle @macandcheese Does this help explain the situation and the reasoning behind the solution that was devised?

**Comment by julio8a** [comment]
@driskull, is this one considered installed too based on https://github.com/Esri/calcite-components/pull/2934?

**Comment by driskull** [comment]
I think so.

**Comment by jcfranco** [comment]
> @macandcheese Does this help explain the situation and the reasoning behind the solution that was devised?

If there's pending work regarding this, can we create a separate issue and close this one? From what I gather, the rest from the original issue has been addressed. Let me know if this isn't the case. cc @benelan

**Comment by macandcheese** [comment]
Yeah I don’t think this is an issue anymore at least for my case above.
